packages/notionable/src/base-trade/sagas/tokens.ts

```diff
@@ -206,7 +206,9 @@ export function availableTokens(
         availableDebtTokens.map((t) => t.id).join(':') !==
           s.availableDebtTokens?.map((t) => t.id).join(':') ||
         availableDepositTokens.map((t) => t.id).join(':') !==
-          s.availableDepositTokens?.map((t) => t.id).join(':');
+          s.availableDepositTokens?.map((t) => t.id).join(':') || 
+        s.debt === undefined || s.collateral === undefined
```

Comment:
Can we move the hasChanged variable lower after `getSelectedToken` is called and change this comparison to `s.debt?.id !== debt.id || s.collateral?.id !== collateral.id`